nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -1131,6 +1135,22 @@ public void run() {
         this.loggingFutures = futures;
     }
 
+    private Long getPidViaRuntimeMXBean(final Logger logger) {
```

Comment:
Not a bad idea - I think the same goes for the method that calls this. This is kind of attempt #2 for getting a pid.